src/main/java/org/cbioportal/web/SurvivalController.java

```diff
@@ -59,18 +60,25 @@ public ResponseEntity<List<ClinicalData>> fetchSurvivalData(
         // prevent reference to this attribute in the swagger-ui interface. this attribute is needed for the @PreAuthorize tag above.
         @Valid @RequestAttribute(required = false, value = "interceptedSurvivalRequest") SurvivalRequest interceptedSurvivalRequest) {
 
+        return new ResponseEntity<>(cachedSurvivalData(interceptedSurvivalRequest),
+                                    HttpStatus.OK);
+    }
+
+    @Cacheable(
+        cacheResolver = "staticRepositoryCacheOneResolver",
```

Comment:
Wondering why we need to use `staticRepositoryCacheOneResolver` at here, I think for the case with lots of permutations like this (`study_ids with sample_ids`) we should use `generalRepositoryCacheResolver`